lib/core/src/data/opt_vec.rs

```diff
@@ -1,5 +1,6 @@
 use crate::prelude::*;
-use super::types::Index;
```

Comment:
I removed the `types::Index` file because I don't like having a separate file which just defines `type Index = usize`. This is not a generic solution. We cannot assume that any Index in the whole codebase is `usize` and such file suggests something like that. Regarding naming - `Index` is definitely much nicer than `Ix`, however it clashes with the `Index` trait, which is a popular one. I'll leave it this way for now, but we need to think more about this naming.